src/backend/cuda/subconfigure.m4

```diff
@@ -62,7 +62,9 @@ if test "$with_cuda" != "no" ; then
     PAC_CHECK_HEADER_LIB([cuda_runtime_api.h],[cudart],[cudaStreamSynchronize],[have_cuda=yes],[have_cuda=no])
     if test "${have_cuda}" = "yes" ; then
         AC_MSG_CHECKING([whether nvcc works])
-        if test -n "$ac_save_CC" ; then
+        # only explicitly set the host compiler if not using the icc Intel
+        # compiler (this conflicts with current versions of nvcc)
+        if test -n "$ac_save_CC" -a "${ac_save_CC}" != "icc"; then
```

Comment:
Did you try `icpc` -- similar to what did for `pgcc` and `nvc`?

Reply:
No, this does not work either. You can simply reproduce this by running:
```console
$ echo '__global__ void foo(int x) {}' | nvcc -x cu -c -ccbin icpc -
```

Instead the more recent `icx` and `icpx` compilers are working (given `-allow-unsupported-compiler` is set; which is probably something we do not want).